app/models/manageiq/providers/workflows/automation_manager/workflow_instance.rb

```diff
@@ -39,7 +39,18 @@ def queue_callback(state, message, _result)
 
   def run
     creds = credentials&.transform_values do |val|
-      ManageIQ::Password.try_decrypt(val)
+      if val.start_with?("$.")
+        name, field = val.match(/^\$\.(?<name>.+)\.(?<field>.+)$/).named_captures.values_at("name", "field")
+
+        # TODO: this isn't filtering out authentication records from other tenants
+        authentication = ext_management_system.authentications.find_by!(:name => name)
+        authentication = Rbac.filtered_object(authentication, :userid => userid, :miq_group => miq_group, :class => Authentication)
+        raise ActiveRecord::RecordNotFound if authentication.nil?
+
+        authentication.send(field)
+      else
+        ManageIQ::Password.try_decrypt(val)
```

Comment:
NOTE now that we allow external credentials do we want to allow inline encrypted credentials?

Reply:
I would prefer that we don't allow any passwords in requests/responses, and that we force this workflow.
But that is probably unrealistic.